### PR TITLE
Add tidy_groups command

### PR DESCRIPTION
**Pull request by elParaguayo**
Moves windows to the leftmost available groups. Layouts and focus can be preserved.

### CODE REVIEW

**Review by m-col — APPROVED** [review]
This seems a bit niche - have you found this to be useful?

Regardless, couple comments but LGTM

File: libqtile/core/manager.py [hunk 1]
```diff
@@ -1600,3 +1600,76 @@ def cmd_get_test_data(self) -> Any:
...
+                return layouts[0]
+            return None
+
+        win = self.current_window
+        empty = []
+
+        for group in self.groups:
+
+            # We need to skip scratchpads
+            if hasattr(group, "dropdowns"):
```
[the remark this answers is not in the document]
Comment (elParaguayo) [reply libqtile/core/manager.py:1638]: I thought about that but when I looked at `self.groups` I get:
```[<group.Group ('1')>, <group.Group ('2')>, <group.Group ('3')>, <group.Group('4')>, <group.Group ('5')>, <group.Group ('6')>, <group.Group ('scratchpad')>]
```
I didn't test `isinstance` though!

**Comment by elParaguayo** [comment]
> This seems a bit niche - have you found this to be useful?
> 
> Regardless, couple comments but LGTM

It was a request on reddit. I can easily post it as an example of a `lazy.function` if we don't want to include it.

It's not something I would use.

**Comment by m-col** [comment]
I think this would be better off in qtile-examples as a dedicated module using lazy.function :)

File: libqtile/core/manager.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-col) [reply libqtile/core/manager.py:1638]: I saw that too but thought something wasn't quite right.... then tried this:
```
> qtile cmd-obj -o cmd -f eval -a "type(self.groups[-1])"
(True, "<class 'libqtile.scratchpad.ScratchPad'>")
```
It inherits `__repr__` from `Group` and then isn't making a new `Scratchpad` method

**Comment by elParaguayo** [comment]
> I think this would be better off in qtile-examples as a dedicated module using lazy.function :)

Ok. Will move.

**Comment by elParaguayo** [comment]
https://github.com/qtile/qtile-examples/pull/75

**Pull request closed without merging**